src/v/debug_bundle/debug_bundle_service.cc

```diff
@@ -222,12 +223,41 @@ ss::future<result<void>> service::initiate_rpk_debug_bundle_collection(
 }
 
 ss::future<result<void>> service::cancel_rpk_debug_bundle(job_id_t job_id) {
+    auto hold = _gate.hold();
     if (ss::this_shard_id() != service_shard) {
         co_return co_await container().invoke_on(
           service_shard,
           [job_id](service& s) { return s.cancel_rpk_debug_bundle(job_id); });
     }
-    co_return error_info(error_code::debug_bundle_process_never_started);
+    auto units = co_await _terminate_mutex.get_units();
+    auto status = process_status();
+    if (!status.has_value()) {
+        co_return error_info(error_code::debug_bundle_process_never_started);
+    } else if (!is_running()) {
+        co_return error_info(error_code::debug_bundle_process_not_running);
+    }
+
+    vassert(
+      _rpk_process,
+      "_rpk_process should be populated if the process has been executed");
```

Comment:
nitpick: this assertion is a bit extraneous if we just checked `process_status()`, no?

Reply:
probably... i guess i'm just trying to be pedantic about it being there... i can remove it

Reply:
meh, no big deal. mainly checking my understanding